packages/e2e-env/config/jest.setup.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import {
+	clearLocalStorage,
+	enablePageDialogAccept,
+	isOfflineMode,
+	setBrowserViewport,
+	switchUserToAdmin,
+	switchUserToTest,
+	visitAdminPage,
+} from '@wordpress/e2e-test-utils';
+
+//Set the default test timeout to 30s
+let jestTimeoutInMilliSeconds = 30000;
+
+// When running test in the Development mode, the test timeout is increased to 2 minutes which allows for errors to be inspected.
+// Use `await jestPuppeteer.debug()` in test code to pause execution.
+if (
+	process.env.JEST_PUPPETEER_CONFIG ===
+	'tests/e2e-tests/config/jest-puppeteer.dev.config.js'
+) {
+	jestTimeoutInMilliSeconds = 120000;
+}
+
+jest.setTimeout( jestTimeoutInMilliSeconds );
+
+/**
+ * Array of page event tuples of [ eventName, handler ].
+ *
+ * @type {Array}
+ */
+const pageEvents = [];
+/**
+ * Set of console logging types observed to protect against unexpected yet
+ * handled (i.e. not catastrophic) errors or warnings. Each key corresponds
+ * to the Puppeteer ConsoleMessage type, its value the corresponding function
+ * on the console global object.
+ *
+ * @type {Object<string,string>}
+ */
+const OBSERVED_CONSOLE_MESSAGE_TYPES = {
+	warning: 'warn',
+	error: 'error',
+};
+
+async function setupBrowser() {
+	await clearLocalStorage();
+	await setBrowserViewport( 'large' );
+}
```

Comment:
This file seems to be something that would be better suited as a part of a `@woocommerce/e2e-utils` package that can then be used for exposing all woocommerce test utilities (including what is currently in woocommerce core). 

I understand the value in plugins just having one package to pull in for setting up an e2e testing environment and all the related utilities, but there's also common use-case where users will want to run e2e tests against their own environment (which they can). Docker is just a convenience. So I wonder if we should follow the pattern established by GB (and elsewhere I've seen) where we have a `@woocommerce/e2e-test-utils` (which extends `@wordpress/e2e-test-utils`) and split out the `woocommerce/env` stuff to its own package (which means it could feasibly evolve for other things including phpunit tests, local dev environment etc).

Reply:
@nerrad I originally thought that it would be a good idea to combine both the Docker setup + e2e utils in one package but I agree with your argument regarding separating them.